src/Parallel/Main.hpp

```diff
@@ -122,6 +124,62 @@ class Main : public CBase_Main<Metavariables> {
       phase_change_decision_data_;
 };
 
+namespace detail {
+
+// the AtSyncIndicator is a silly hack to have a centralized indication to start
```

Comment:
How about expanding a bit here something like "Charm++ AtSync effectively requires an additional global sync to the quiescence detection we do for switching phases. However, AtSync only needs to be called an one array, so the AtSyncIndicator..."? I remember this from your explanation on a call, but I'm not sure I would've figured it all out from what you have written here so far :)